modules/core/test/test_quaternion.cpp

```diff
@@ -258,6 +258,193 @@ TEST_F(QuatTest, interpolation){
     EXPECT_EQ(Quatd::spline(tr1, tr2, tr3, tr3, 0.5), Quatd(0.336889853392, 0.543600719487, 0.543600719487, 0.543600719487));
 }
 
+static const char* angleTypeName[24] = {
+    "INT_XYZ", "INT_XZY", "INT_XZX", "INT_XYX",
+    "INT_YXZ", "INT_YZX", "INT_YXY", "INT_YZY",
+    "INT_ZXY", "INT_ZYX", "INT_ZXZ", "INT_ZYZ",
+
+    "EXT_XYZ", "EXT_XZY", "EXT_XZX", "EXT_XYX",
+    "EXT_YXZ", "EXT_YZX", "EXT_YXY", "EXT_YZY",
+    "EXT_ZXY", "EXT_ZYX", "EXT_ZXZ", "EXT_ZYZ"
+};
+
+TEST_F(QuatTest, EulerAngles){
+    Vec3d test_angle={0.523598,0.78539,1.04719};
+    Quatd qEuler0 = Quatd(0, 0, 0, 0);
```

Comment:
Why not transform these huge lists (here and below) to loops?

Reply:
OK, thanks, I solved this problem by using the for loop method and array `qEuler`.